pkg/kubelet/kubelet.go

```diff
@@ -2421,6 +2424,46 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}
 }
 
+func (kl *Kubelet) syncOSAndArchLabelsOnce() (*v1.Node, error) {
+	node, err := kl.GetNode()
+	if err != nil {
+		return nil, fmt.Errorf("error getting node: %v", err)
+	}
+	newNode := node.DeepCopy()
+	// Set the OS label if there is mismatch
+	osName, ok := newNode.Labels[v1.LabelOSStable]
+	if !ok || osName != sysruntime.GOOS {
+		if len(newNode.Labels) == 0 {
+			newNode.Labels = make(map[string]string)
+		}
+		newNode.Labels[v1.LabelOSStable] = sysruntime.GOOS
+	}
+	// Set the arch if there is a mismatch
+	arch, ok := newNode.Labels[v1.LabelArchStable]
+	if !ok || arch != sysruntime.GOARCH {
+		if len(newNode.Labels) == 0 {
+			newNode.Labels = make(map[string]string)
+		}
+		newNode.Labels[v1.LabelArchStable] = sysruntime.GOARCH
+	}
+	updatedNode, patchBytes, err := nodeutil.PatchNodeWithStatusAndLabels(kl.kubeClient.CoreV1(), node.Name, node, newNode)
```

Comment:
I don't think we should be creating a new kubeClient here, don't we have one already?

Reply:
I didn't fully understand. Where we have `CoreV1Interface` object that I can re-use. I couldn't find it in the kubelet [struct](https://github.com/kubernetes/kubernetes/blob/ada53f25a6c807b835efcf08f4bbbe9257ff5e7a/pkg/kubelet/kubelet.go#L893), however I could have missed it.